Season Config Optimizations

harishbommakanti opened the pull request:
- code cleanup like deleting unnecessary nesting, adding return statements
- changed method of storing elements from x elements producing x cookies to only one cookie with an object holding the elements. Updated all of the code in other pages like teamData that rely on the elements cookie to work with the new method.
- added an input type box along with element name/point value to help later in match scouting

harishbommakanti committed:
Create CNAME file

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Minor change in the nav list html code

- home.html --> index.html

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Merge branch 'master' into gh-pages

harishbommakanti committed:
Initial Commit: Set up Jekyll environment

- installed rubyinstaller/ruby/gem/jekyll and set up all the jekyll environment

harishbommakanti committed:
Created gitignore, removed jekyll-cache and _site/ files

- gitignore updated to ignore cache and site output files
- removed cache and site output files

harishbommakanti committed:
Created nav.html file

- added nav.html to file, will start to implement it in the pages

harishbommakanti committed:
Used jekyll include for nav button

-  added nav button html code to nav.html and used liquid include tags to reference it

harishbommakanti committed:
Updated nav.html: added liquid comment

harishbommakanti committed:
Created head.html

- copied all of the css/js/font sheets to an includes head.html file

harishbommakanti committed:
Used jekyll include head.html on all pages

- used jekyll include head.html on all pages
- fixed a minor bug for button styling

harishbommakanti committed:
Created a basic layout file/directory

harishbommakanti committed:
Added code to the basic.html layout

- has the template for the basic layout on all pages
- modified season page to make its body tag similar to others (instead of onchange, I just added that to the doc.onReady in the js file)

harishbommakanti committed:
Used basic.html as a layout on all pages

- used basic.html as a layout on all html pages to avoid copying over code ie increasing possibility of buggy code

harishbommakanti committed:
minor fix: deleting commented out code

- deleted old commented out html code

harishbommakanti committed:
PR fix: updated gitignore file

will change the navbar to be better on mobile devices

harishbommakanti committed:
Added css/html for a navbar

- used w3css and my own styling to go from a navbar button to an actual navbar
- deleted navbar button code (js/html/css)
- will now work on creating an active tab

harishbommakanti committed:
Added more styling to navbar

harishbommakanti committed:
PR fix: re-added links to different pages on index.html

harishbommakanti committed:
PR fix: updated styles.css for a more efficient way of making all elements ubuntu maroon color

harishbommakanti committed:
Created pages.yaml, added _data folder

- added file to make the collapsable navbar on smaller screens easier to use with jekyll templating

harishbommakanti committed:
Navbar more friendly to smaller screens

- now the navbar is collapsable on smaller screens

harishbommakanti committed:
PR fix: moved title to navbar

- moved title as part of the red navbar chunk as it looks a little better

harishbommakanti committed:
PR fix: removed commented code in basic.html

harishbommakanti committed:
PR fix: simplified html of navbar

- moved the title into the navbar div

harishbommakanti committed:
PR fix: meta tag edited, some navbar styling

- meta tag edited so the site renders well on the chrome editor

harishbommakanti committed:
PR fix: title and menu are on same line

- added an extra title div that adjusts so its on the same line with the menu button, makes all the home/season config... buttons centered while not making the navbar look too bad

harishbommakanti committed:
Removed github/instagram/twitter links from main page

- removed html/css for social media links on front page as they are 1) not really linked to frc scouting and 2) kind of hard to style across screen sizes

harishbommakanti committed:
Further title styling, removed some uneccesary text on index.html

harishbommakanti committed:
PR fix: separated meta tags

harishbommakanti committed:
PR fix: removed unecessary div tag from team-data.html

harishbommakanti committed:
Initial Commit of Team Data Opmtimizations PR: simplified pit q/a storing method

- simplified way of storing/showing pit questions/answers (look into the github project card for more info)

harishbommakanti committed:
Match data storing/showing is also simpler

- array of len 1 dictionaries --> 1 dictionary of multiple key/value pairs

harishbommakanti committed:
Updated comments on importing/exportin

- due to the last 2 changes (array of dictionaries to a straightforward dictionary) I'm just adjusting the comments so that ppl won't be confused in the future

harishbommakanti committed:
Added a custom notes section for team data

- added a notes section for the person to include info like 'tippy' or 'doesn't cooperate with teamates well' to be defining factors during gameplay

harishbommakanti committed:
Changed radio list to select

- radio list to select as events will have upwards of 20-30 teams, don't want to have to scroll through all of that

harishbommakanti committed:
Updated loadPit() for new pit method

- a bit late of a commit, but I forgot that this was a thing...
- loading previous answers for the pit input boxes now works with the simpler storage method

harishbommakanti committed:
Fixing bug with loading previous pit answers

- fixed basically a nullpointer exception

harishbommakanti committed:
Changed th pit table to w3-grid

- changed the pit table to w3-grid to be better formatted on mobile
- needs further styling and I need to revise code for reading the table since the layout changed

harishbommakanti committed:
Revised savePit() and loadPit() to make sure they worked with textarea and w3-grid

- revised other loading/saving methods to make sure they read from textareas instead of inputs, and w3-rows instead of tables

harishbommakanti committed:
Added autosize.js from Jack Moore's github profile

harishbommakanti committed:
Finished using autosize.js forthe pit textareas

- realized that I needed the whole github project and not just the autosize.js, put the autosize-master under the scripts folder
- used the autosize method in loadpit() and it works fine

harishbommakanti committed:
Merge branch 'master' into Optimizations

harishbommakanti committed:
PR fix: simplified newTRHtml code with css

harishbommakanti committed:
PR fix: Custom Notes not required to save match answers

harishbommakanti committed:
Pit table changed to w3-grid

- removed code for having the allTeamsPitTable being a table
- used a w3-grid
- added some css

harishbommakanti committed:
added a css class to wrap the div contents within the div

harishbommakanti committed:
PR fix: Interim switch bewteen w3-grid layouts

- change to header on top to header on left for the w3-grid for the allTeamsPitAnswers table
- still need to normalize heights/widths and add a scrollbar, hence the interim part

harishbommakanti committed:
PR fix: w3-grid layout fixed: questions are on the left side

- only fix to be done on this part now is to have a scrollbar for horizontal scrolling
- added some css for styling

harishbommakanti committed:
PR fix: reverted allPitTable w3-grid to an actual table, but 'rotated' the table left (header on left etc)

- reverted to a table format after having issues with width and standardizing rows
- added a horizontal scrollbar
- will now work on doing the same with the matchDataTable

harishbommakanti committed:
PR fix: cumulative match data table also re styled

- cumulative match data table is also styled and is given a max width: will show a horizontal scroll bar when the # of teams makes the table stretch out too far to the right

harishbommakanti committed:
PR fix: minor code fixes

- fixes such as changing textarea width and flipping conditional checking

harishbommakanti committed:
PR fix: edited way to append td's in the table

- instead of abusing class usage I used 'question' and 'element' attributes, allowing me to use the question itself instead of taking the spaces and other nonalphanumerica questions off
- edited small bug for string comparison: == instead of ===

harishbommakanti committed:
PR fix: added method for sanitizing HTML

- added method for sanitizing an HTML line, still need to implement it

harishbommakanti committed:
PR fix: implemented the sanitizeHTMLLine method

- implemented the method in the pit table/match table code

harishbommakanti committed:
Season Config Optimizations Initial Commit: Optimized element storing in cookie

- before, there were x cookies if there were x elements saved. Now, there is only 1 cookie for elements for each season with an object with key-value pairs for the element and the points
- need to work through ALL the code in teamData and importExport to work with this now

harishbommakanti committed:
Updated import/export page to work with the new element storing method

- export/import now works with the way of storing elements in one cookie rather than multiple

harishbommakanti committed:
Updated teamData page to work with new way of storing elements

- instead of having to iterate through all cookies to find element cookies, the program finds the one cookie with the object with the element/point values

harishbommakanti committed:
Added/modified html to add an input type for the elements

- input type for elements will help for displaying inputs in match data questionnaire

harishbommakanti committed:
Added code for storing input type for elements

- added code for storing the input type
- points and the input type are now in a nested object for each element, need to work on updating import/export and teamData for this now

harishbommakanti committed:
Match Data input fields have types based on what user selected in season config

- now, if the element was made with an input type of numeric, the input would have a 'number' type, vice versa for text

harishbommakanti committed:
Code cleanup, moving pit method outside document.ready()

- updated delete element method to work with object
- moved really large function for save pit question outside of document.ready
- cleaned up code for save pit questions

harishbommakanti committed:
More code cleanup

- deleting indentations and stuff like that

harishbommakanti committed:
Merge branch 'master' into Optimizations

mshafer1 commented on season-config.html:
```diff
@@ -15,24 +15,31 @@ <h3 id="newSeasonConfirmation"></h3><br><br>
-    integers for the points.<br><br>
+<h3>Enter in any scoring opportunities in the first text box and how many points that scoring opportunity is worth in the second text box.
+    <br><br>
+    Also, please enter the type of answer to be expected when answering questions about the scoring opportunity in 
```
Why do we need the "points worth"?

harishbommakanti replied:
I just left it there at the start in case someone wanted to improve the app and make a feature like team 1 scores 20 pts per average in a certain event or things like that. I didn't implement that feature because I thought it'd involve interpreting text answers ranging from 'yes' to 'maybe, only sometimes' and stuff like that. So just as groundwork for a possible future feature

mshafer1 commented on scripts/teamDataScript.js:
```diff
@@ -319,21 +319,23 @@ function loadMatch()
+        var inputType = nestedObject['inputType'];
 
-        var inputHTML = `<input type="number" class="match-frequency js_clear_on_load">`;
+        var inputHTML = `<input type="${(inputType==='numeric')?'number':'text'}" class="match-frequency js_clear_on_load">`;
```
we might want to use `textarea` in place of the `text` and setup the auto sizing.

mshafer1 commented on scripts/teamDataScript.js:
```diff
@@ -319,21 +319,23 @@ function loadMatch()
-        var inputHTML = `<input type="number" class="match-frequency js_clear_on_load">`;
+        var inputHTML = `<input type="${(inputType==='numeric')?'number':'text'}" class="match-frequency js_clear_on_load">`;
         var elementText = `${element}`;
         var newTRHtml = `<tr><td class='match-element scoutingTableRow'>${elementText}</td><td class='scoutingTableRow'>${inputHTML}</td></tr>`;
```
these inputs are winding up very arrow, consider using a width="80%" on the input.

mshafer1 commented on season-config.html:
```diff
@@ -15,24 +15,31 @@ <h3 id="newSeasonConfirmation"></h3><br><br>
         <tr>
             <th><b>Scoring Method</b></th> 
-            <th><b>Points Worth</b></th> 
+            <th><b>Points Worth</b></th>
+            <th><b>Input Type</b></th> 
```
Did you test import/export?

harishbommakanti replied:
Yeah, seemed to work fine. Since all the elements are now in one object (changed in this PR), and the import/export just reads/reports the content of the object (I changed it to do this when reading/reporting elements in this PR) it imports/exports cleanly

mshafer1 commented on scripts/seasonConfigScript.js:
```diff
@@ -46,20 +42,19 @@ function addNewSeason()
+    }
+
+    var seasonList = getCookie('seasonList').split(SEASON_LIST_SEPARATOR);
+    //console.log('seasonList:' + seasonList);
```
don't check in commented out code, use source control to track having it in the past and why you got rid of it.

mshafer1 reviewed (changes requested):
like the ability to do text input.

If you're going to always add the "Custom Notes", then you should them that in the setup page ("a notes region will be provided as well" or something.

Horizontal scrolling of the tables...I'm just seeing them squish and fit on the page...

harishbommakanti committed:
PR fix: input text boxes --> textareas

- changed inputs to textareas in case of text input, used autosize for all boxes
- fixed bug where deleteSpecificSeason and deleteSpecificEvent weren't updated with recent cookie storing

harishbommakanti committed:
PR fix: resizing on match input areas

- added a width: 80% on the sizing for match inputs
- changed custom notes input to a textarea

mshafer1 commented on scripts/teamDataScript.js:
```diff
@@ -316,7 +316,7 @@ function loadMatch()
     var cookieList = document.cookie.split(';');
 
-    var matchInputHTML = `<input type="number" class="matchNumInput match-frequency js_clear_on_load">`;
+    var matchInputHTML = `<input style = "width: 80%" type="number" class="matchNumInput match-frequency js_clear_on_load">`;
```
this works great - apparently this is how you need to do it for the pit answers too.

harishbommakanti committed:
PR fix: applied 80% width to pit textareas too

- used width=80% for the textareas for the pit questions

harishbommakanti committed:
Edited HTML of the season config page

- edited HTML of season config page to reflect feedback from the team

harishbommakanti committed:
Added 'Ranking Points' section to Match Data on the Team Data page

- users can also record ranking points scored by a team as requested by the team feedback

mshafer1 commented on season-config.html:
```diff
@@ -15,14 +15,16 @@ <h3 id="newSeasonConfirmation"></h3><br><br>
+    ● For scoring opportunities like elements and end game actions, enter in their name in the first box and how many points the
+    action is worth in the second box.<br>
+    ● For the input type, enter in 'Numeric' (Users can enter in 0 for things like "no endgame climb").<br><br>
+    ● If you would like to also enter in questions here to record specific in-game characteristics of a team 
```
For scoring opportunities like elements and end game actions, enter in their name in the first box and how many points the action is worth in the second box.

For the input type, choose 'Numeric' to allow users to enter a number value and 'Text' to allow users to enter arbitrary text.

mshafer1 commented on scripts/teamDataScript.js:
```diff
@@ -345,7 +345,10 @@ function loadMatch()
     var customNotesInputHTML = `<textarea class="match-textAreas match-frequency js_clear_on_load"></textarea>`; //for custom notes each match
     var customNotesRow = `<tr> <td class="match-element scoutingTableRow">Custom Notes</td> <td class="scoutingTableRow">${customNotesInputHTML}</td> </tr>`;
+    var RPInputHTML = `<input style = "width: 80%" type="number" class="match-frequency js_clear_on_load">`; //for ranking points each match
+    var RPRow = `<tr> <td class="match-element scoutingTableRow">Ranking Points</td> <td class="scoutingTableRow">${RPInputHTML}</td> </tr>`;
```
Not everyone is going to want to "track" ranking points (they can see who's ranked highest through lots of other means) - we should probably just tell users who want this to utilize the "Numeric" type input.

harishbommakanti committed:
Fixed text on season-config

- changed some of the instruction text to be more simple and to the point

harishbommakanti committed:
Removed ranking points feature

- removed ranking points feature as it can be easily found many other places during the competition: if needed, it can be added as a thing to track during the event
